### ISSUE TITLE
rocJM producing unexpected effects

### ISSUE THREAD

**Issue by DocEd** [drizopoulos/JM#20]
Hi @drizopoulos,

I am fitting some joint models of organ dysfunction in critical illness. In order to help with stability, I center and scale variables before modelling (including the outcome of the longitudinal submodel). Some of the values of the longitudinal outcome will now take negative values. There is a strange appearance to the ROC plots (see reproducible example from the package sample data) where the curves don't complete their path. The AUC results are also much lower than would be expected. I can't seem to isolate the issue to any other cause. Is this normal behaviour? Must variables inside the joint model only take positive values? is it necessary that other variables in the model are also positive (e.g. explanatory variables inside the longitudinal or cox submodels)?

```r
library(JM)

pbc2$status2 <- as.numeric(pbc2$status != "alive")
pbc2.id$status2 <- as.numeric(pbc2.id$status != "alive")

# center and scale longitudinal outcome
bil_scaling <- scale(pbc2$serBilir)

# reassign to data frame
pbc2$serBilir <- bil_scaling[, 1]
pbc2.id$serBilir <- pbc2$serBilir[pbc2$year == 0]

# fit models
lmeFit <- lme(serBilir ~ ns(year, 3),
              random = list(id = pdDiag(form = ~ ns(year, 3))), data = pbc2)
survFit <- coxph(Surv(years, status2) ~ drug, data = pbc2.id, x = TRUE)

jointFit <- jointModel(lmeFit, survFit, timeVar = "year", 
                       method = "piecewise-PH-aGH")

# ROC plots
ND <- pbc2[pbc2$id == "7", ]
roc1 <- rocJM(jointFit, dt = c(2, 4, 8), data = ND, idVar = "id")
plot(roc1)
```

![Screenshot 2020-09-01 at 15 37 11](https://user-images.githubusercontent.com/17110713/91865221-10744d00-ec69-11ea-942d-8b2d267a0635.png)

I couldn't find any other reference to this in your book or on stack overflow. Thank you in advance for your time and insights on this.

**Comment by DocEd** [comment]
I've done a little more investigation Prof @drizopoulos .
I've found where the problem originates. When choosing the limits for cut off thresholds of the longitudinal biomarker the following lines of code in `JM::rocJM` do not anticipate a negatively valued biomarker:

```r
if (is.null(cc) || !is.numeric(cc)) {
  pc <- quantile(object$y$y, c(0.05, 0.95), names = FALSE)
    if (is.null(min.cc) || !is.numeric(min.cc)) 
      min.cc <- min(min.cc, min(object$y$y) - pc[1]) # <- here is the problem
    if (is.null(max.cc) || !is.numeric(max.cc)) 
      max.cc <- max(max.cc, max(object$y$y) + pc[2])
  cc <- seq(min.cc, max.cc, length.out = 251)
}
```

Because the 0.05 quantile is also negative, it results in a lower limit for the threshold that is higher than the minimum value of the biomarker. This is easy to override with the `min.cc` argument, but is it worth changing the default behaviour to prevent this happening? I believe the following would work without causing any unintentional side effects:

`min.cc <- min(min.cc, min(object$y$y) - abs(pc[1]))`

I can see that `pc` is not used anywhere else. I'm happy to submit a pull request if you think this is appropriate.

This has also uncovered another interesting issue. The ROC curve appears to be sensitive to the scale of the time variable. When I scale time to be on [0, 1] to help with convergence, the ROC curve is flattened. See below

![roc_problem](https://user-images.githubusercontent.com/17110713/92220012-4268f780-ee93-11ea-914f-f4470e82f245.png)

I have rescaled the time variable in this joint model (the same as the previous example) to be in years, months and on [0, 1]. Note how the curve is sensitive to the scaling. Though I cannot see a good explanation at the moment as to why this would be.
